x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/vulnerabilities.tsx

```diff
@@ -116,7 +116,12 @@ const VulnerabilitiesContent = ({ dataView }: { dataView: DataView }) => {
   const slicedPage = usePageSlice(data?.page, pageIndex, pageSize);
 
   const invalidIndex = -1;
-  const selectedVulnerability = data?.page[urlQuery.vulnerabilityIndex];
+
+  const selectedVulnerability = useMemo(() => {
+    return slicedPage?.find(
```

Comment:
@Omolola-Akinleye why we use find if we know the index?
shouldn't slicedPage[urlQuery.vulnerabilityIndex] would work here? and would be O(1) instead of O(n)

Reply:
Fair point. Typescript throw undefined when I was using  `data?.page`.  I incorrectly confused slicedPage for `data. page` as being possibly undefined. I will adjust that.